server/src/internalClusterTest/java/org/opensearch/search/aggregations/bucket/GlobalIT.java

```diff
@@ -87,6 +87,7 @@ public void testWithStatsSubAggregator() throws Exception {
         SearchResponse response = client().prepareSearch("idx")
             .setQuery(QueryBuilders.termQuery("tag", "tag1"))
             .addAggregation(global("global").subAggregation(stats("value_stats").field("value")))
+            .setProfile(randomBoolean())
```

Comment:
The test should probably very that results are now correct (vs incorrect results before the fix).

Reply:
Using [randomized testing for coverage](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#use-randomized-testing-for-coverage) is generally an anti-pattern. Probably better to explicitly test both cases of `setProfile`.

Reply:
@andrross I thought about it but the post mentions `For instance the number of shards should not impact date_histogram aggregations, and the choice of the store type (niofs vs mmapfs) does not affect the results of a query. Such randomization helps improve confidence that we are not relying on implementation details of one component or specifics of some setup.` 

In this case since profiling should not change the results of query but returns extra information in the response I thought randomization should be fine

Reply:
My thinking is that at the aggregation layer, something like store type is completely abstracted because all interaction is done through something like a generic directory interface and the aggregation layer has no knowledge of the specific implementation. In that case randomization is fine because it's essentially just verifying nothing is leaking through the abstraction layers. In the profiling case there is now a specific if condition in the aggregation code, so both paths should be explicitly tested.

Reply:
@andrross Have updated the test to explicitly add for both the profile enabled/disabled case